p2p/src/types.rs

```diff
@@ -161,6 +166,45 @@ impl Readable for PeerAddr {
 	}
 }
 
+impl<'de> Visitor<'de> for PeerAddrs {
```

Comment:
I'm *not sure* you need to implement the deserializer manually. Why not simply add the serde `Deserialize` trait to `PeerAddrs`?

Reply:
Because it's not just deserializing, its resolving the DNS name into one or many IP addresses by calling to_socket_addrs(). Serde wouldn't know how to do that as far as I'm aware.

Reply:
👍